File: Python/LRUCache.py
class LRUCache:
    
    def __init__(self, capacity: int):
        self.a=[]
        self.d={}
        self.cap=capacity  

    def get(self, key: int) -> int:
        if(key in self.a):
            self.a.remove(key)
            self.a.insert(0,key)
            #print("get",self.a)
            return self.d[key]
        
        return -1

    def put(self, key: int, value: int) -> None:
        
        if key in self.d:
            if key in self.a:
                self.a.remove(key)
        
        if (len(self.a)==self.cap):
                self.a.pop()
        
        self.a.insert(0,key)
        
        self.d[key]=value
        #print("put",self.a)
        


# Your LRUCache object will be instantiated and called as such:
# obj = LRUCache(capacity)
# param_1 = obj.get(key)
# obj.put(key,value)

File: Javascript/LRU_Cache.js
/*
 * Problem link: https://leetcode.com/problems/lru-cache/
 * Optimized for O(1), Get/Set Option
 * */
/**
 * @param {number} capacity
 */
var LRUCache = function(capacity) {
    this.map = new Map();
    this.capacity = capacity;
};
/**
 * @param {number} key
 * @return {number}
 */
LRUCache.prototype.get = function(key) {
    if(this.map.has(key)){
        const value = this.map.get(key);
        this.map.delete(key);
        this.map.set(key, value)
        return this.map.get(key);
    }else{
        return -1;
    }
};
/**
 * @param {number} key
 * @param {number} value
 * @return {void}
 */
LRUCache.prototype.put = function(key, value) {
    if(this.map.size < this.capacity){
        this.map.set(key, value);
    } else {
        this.map.delete(this.map.keys().next().value);
        this.map.set(key, value);
    }
};
/**
 * Your LRUCache object will be instantiated and called as such:
 * var obj = new LRUCache(capacity)
 * var param_1 = obj.get(key)
 * obj.put(key,value)
 */